timemachine/fe/mle.py

```diff
@@ -104,3 +104,96 @@ def loss(x):
 
     # arbitrary additive constant
     return result - result[0]
+
+
+def _bootstrap_node_vals(
+    edge_idxs,
+    edge_diffs,
+    edge_stddevs,
+    ref_node_idxs,
+    ref_node_vals,
+    ref_node_stddevs,
+    n_bootstrap=100,
+    seed=0,
+):
+    """call infer_node_vals multiple times with Gaussian bootstrapped edge_diffs, ref_node_vals"""
+
+    if len(ref_node_idxs) == 0:
+        print("no reference node values: picking node 0 as arbitrary reference")
+        ref_node_idxs = np.array([0], dtype=int)
+        ref_node_vals = np.array([0], dtype=float)
+        ref_node_stddevs = np.array([0], dtype=float)
+
+    n_edges = len(edge_idxs)
+    n_nodes = len(set(edge_idxs.flatten()))
+    n_refs = len(ref_node_idxs)
+
+    rng = np.random.default_rng(seed)
+
+    def estimate_node_val_w_offset(edge_vals, node_vals):
+        estimated_node_vals = infer_node_vals(edge_idxs, edge_vals, edge_stddevs)
+        estimated_node_offset = np.mean(node_vals - estimated_node_vals[ref_node_idxs])
+        return estimated_node_vals + estimated_node_offset
+
+    bootstrap_estimates = np.zeros((n_bootstrap, n_nodes))
+
+    for i in range(n_bootstrap):
+        noisy_edge_diffs = edge_diffs + rng.standard_normal(n_edges) * edge_stddevs
```

Comment:
note: this part assumes gaussian edge error and is coupled to the definition above `edge_lls = norm.logpdf`

if we switch to heavy-tailed or other edge likelihoods as suggested here https://github.com/proteneer/timemachine/pull/399#issuecomment-826850602 , this line should be updated, e.g. to `rng.standard_t`...

Reply:
Not clear what 'definition above' you are referring to. Don't see an `edge_lls`

Reply:
https://github.com/proteneer/timemachine/blob/71a963a69329836bdd871b791cf6202b5b97e0f2/timemachine/fe/mle.py#L41 , introduced in https://github.com/proteneer/timemachine/pull/932

Reply:
Thanks for clarifying

Reply:
Can you add a comment here and at line 41 to make sure these are synchronized?

Reply:
Applied in https://github.com/proteneer/timemachine/pull/941/commits/d09177cb6e566dc0bd340ae4b9b4cbe0261f5ca2